pkg/controller/federatedtypeconfig/controller.go

```diff
@@ -266,7 +278,7 @@ func (c *Controller) ensureFinalizer(tc *corev1a1.FederatedTypeConfig) error {
 	}
 	finalizers.Insert(finalizer)
 	accessor.SetFinalizers(finalizers.List())
-	_, err = c.client.FederatedTypeConfigs(tc.Namespace).Update(tc)
+	_, err = c.client.FederatedTypeConfigs(tc.Namespace).UpdateStatus(tc)
```

Comment:
This change will prevent the finalizer from being saved.  Everything but status is written with `Update`, and only status is written with `UpdateStatus`.

Reply:
@sohankunkerkar were you able to confirm if this works? Otherwise, per @pmorie we may have a bug with CRD.